CONTRIBUTING.md

```diff
@@ -84,157 +84,160 @@ you can mark the language as optional (by putting it between parenthesis).
+
+@attribute        ; attribute annotations (e.g. Python decorators)
+@property         ; the key in key/value pairs
 ```
 
 #### Functions
 
-```scheme
-@function         ; function definitions
-@function.builtin ; built-in functions
-@function.call    ; function calls
-@function.macro   ; preprocessor macros
+```query
+@function             ; function definitions
+@function.builtin     ; built-in functions
+@function.call        ; function calls
+@function.macro       ; preprocessor macros
 
-@method           ; method definitions
-@method.call      ; method calls
+@function.method      ; method definitions
+@function.method.call ; method calls
 
-@constructor      ; constructor calls and definitions
-@parameter        ; parameters of a function
+@constructor          ; constructor calls and definitions
+@operator             ; symbolic operators (e.g. `+` / `*`)
 ```
 
 #### Keywords
 
-```scheme
-@keyword             ; various keywords
-@keyword.coroutine   ; keywords related to coroutines (e.g. `go` in Go, `async/await` in Python)
-@keyword.function    ; keywords that define a function (e.g. `func` in Go, `def` in Python)
-@keyword.operator    ; operators that are English words (e.g. `and` / `or`)
-@keyword.return      ; keywords like `return` and `yield`
-
-@conditional         ; keywords related to conditionals (e.g. `if` / `else`)
-@conditional.ternary ; ternary operator (e.g. `?` / `:`)
-
-@repeat              ; keywords related to loops (e.g. `for` / `while`)
-@debug               ; keywords related to debugging
-@label               ; GOTO and other labels (e.g. `label:` in C)
-@include             ; keywords for including modules (e.g. `import` / `from` in Python)
-@exception           ; keywords related to exceptions (e.g. `throw` / `catch`)
+```query
+@keyword                   ; keywords not fitting into specific categories
+@keyword.coroutine         ; keywords related to coroutines (e.g. `go` in Go, `async/await` in Python)
+@keyword.function          ; keywords that define a function (e.g. `func` in Go, `def` in Python)
+@keyword.operator          ; operators that are English words (e.g. `and` / `or`)
+@keyword.import            ; keywords for including modules (e.g. `import` / `from` in Python)
+@keyword.storage           ; modifiers that affect storage in memory or life-time
+@keyword.repeat            ; keywords related to loops (e.g. `for` / `while`)
+@keyword.return            ; keywords like `return` and `yield`
+@keyword.debug             ; keywords related to debugging
+@keyword.exception         ; keywords related to exceptions (e.g. `throw` / `catch`)
+
+@keyword.conditional         ; keywords related to conditionals (e.g. `if` / `else`)
+@keyword.conditional.ternary ; ternary operator (e.g. `?` / `:`)
+
+@keyword.directive         ; various preprocessor directives & shebangs
+@keyword.directive.define  ; preprocessor definition directives
 ```
 
-#### Types
+#### Punctuation
 
 ```scheme
```

Comment:
This is later changed in e33e9461d51d43b59c1d983af901aa15bc103d35, but I think it should be changed in this commit as well

Reply:
I don't think that's easily possible, unless I either squash commits that shouldn't be squashed or completely re-stage all these changes. (Remember that I need to rebase `main` on top of this PR, and that is easier if the commits are cherry-picked exactly.)

Reply:
Then i'm fine with the current revision. Also easier to view diffs with `main`